Explorer/Assets/Scripts/CrdtEcsBridge/JsModulesImplementation/EngineAPIImplementation.cs

```diff
@@ -256,7 +264,16 @@ private void SerializeOutgoingCRDTMessages(IReadOnlyCollection<ProcessedCRDTMess
             if (outgoingMessages.Count == 0) return;
 
             foreach (ProcessedCRDTMessage processedCRDTMessage in outgoingMessages)
+            {
+                OutgoingCRDTMessages.Add(processedCRDTMessage);
                 crdtSerializer.Serialize(ref span, in processedCRDTMessage);
+            }
+        }
+
+        private async UniTaskVoid DisposeMemoryOnNextFrame(IMemoryOwner<byte> memoryOwner)
```

Comment:
It's super cheaty and can lead to random pooling problems, we should come up with something else

Reply:
Yeah I know, it's the worst part of the whole branch, but I didn't find any other way at the moment.

I implemented that hack to be able to move forward ASAP.

I'll try re-thinking the approach after I have all the observables working in the branch, if you can think of any other approach I'm all eyes XD

Reply:
I've adopted a different less-hacky approach at https://github.com/decentraland/unity-explorer/pull/662/commits/a3c71f1e8a29d04801ab9aac9585dcb2486f8c83

I minimized the modifications done to `EngineAPIImplementation` by moving the `APPEND` and `AVATAR_EMOTE_COMMAND` filtering at child class `SDKObservableEventsEngineAPIImplementation`. And the message disposal is triggered by `SDKObservableEventsEngineApiWrapper` after reading the memory to populate the observer [here](https://github.com/decentraland/unity-explorer/blob/a3c71f1e8a29d04801ab9aac9585dcb2486f8c83/Explorer/Assets/Scripts/SceneRuntime/Apis/Modules/EngineApi/SDKObservableEvents/SDKObservableEventsEngineApiWrapper.cs#L144~L159)